text/0000-add-ansi-escape-codes.md

```diff
@@ -0,0 +1,63 @@
+- Feature Name: add_ansi_escape_codes
+- Start Date: 2022-02-14
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the addition of more escape code functions to the ANSI primitive of the term package and the consolidation of the thirty-two existing color functions into four functions.
+
+# Motivation
+
+The purpose of these additional methods is to cover more of the standard ANSI escape codes within the stdlib.
+
+# Detailed design
+
+- Add escape codes for 3-bit, 4-bit, 8-bit, and 24-bit colors for the foreground and background.  
+The naming scheme of the color functions would be `{fg,bg}_{3,4,8,24}bit`.  
+The 3-bit, 4-bit, and 8-bit versions would take a U8 specifying the color.  
+The 24-bit versions would take three U8s specifying the r, g, and b values.  
+- Remove the thirty-two named color functions in favor of the 3-bit and 4-bit functions.
+- Add escape codes for enter and leave alternate screen buffer.
+- Add escape code for the bell alert.
+- Change `clear` and `erase` functions to take a U8 specifying what part of the screen/line to clear.  
+The default for the escape codes if a value isn't given is zero.  
+The action these functions currently take is three for `clear` and one for `erase`.  
+I would like to have the default value in these functions be zero.
+- Add escape codes for scroll up/down.
+- Add escape codes for next/previous line.
+- Add escape code for cursor horizontal position (moves the cursor to column n in the current line).
+- Add escape codes for hide/show cursor.
+- Add escape codes for save/restore cursor position.
+- Add escape code for get cursor position (it is sent to stdin).
+- Add escape codes for various text styles: italic, strike, conceal, faint.
```

Comment:
Would you do me a favor and add the codes you mean here? I understand some from looking them up, but others I am not seeing in reference tables.

Reply:
Do you think it would be better to have them listed after each code is mentioned like this:

- Add escape code for hide cursor `\x1B[?25l` and show cursor `\x1B[?25h`.

Or just have the table row from the escape codes page like this:

Code | Abbr | Name | Effect
-- | -- | -- | --
CSI ? 25 h | DECTCEM |   | Shows the cursor, from the VT220.
CSI ? 25 l | DECTCEM |   | Hides the cursor.
 

Reply:
Good question. I would list them inline, personally, as I do not think it breaks up the flow of the bullet points.